keps/sig-node/4800-cpumanager-split-l3cache/README.md

```diff
@@ -0,0 +1,680 @@
+- It does not change behavior of non-static policy
+- It preserves the behavior of other static options.
+
+## Design Details
+
+To make Kubelet be uncore cache aware for CPU sensitive workloads the CPU allocation will require the addition of the `UnCoreCacheID` to enhance CPU scheduling for Pods. This creates the need for updating the `CPUInfo` and the `CPUToplogy` data structures to include `UnCoreCacheID`, which is provided by `cAdvisor` during the CPU discovery process.
+
+A new CPUManager policy option will be introduced, `align-cpus-by-uncorecache`, to enable this feature. With this feature enabled, a modification of the existing default scheduling function is needed to take into consideration of a three level hierarchy: Socket, NUMA, and UnCoreCache. The allocation process will require to attempt to allocate full sockets. Then allocation should attempt the full NUMA nodes. Then allocation must consider the UnCoreCache. Since this policy augments the CPUManager's Static policy, this feature will only be applied to guaranteed pods with whole CPU requests.
+
+The algorithm for align-cpus-by-uncorecache will be implemented to follow the default packed behavior of the existing static CPUManager allocation with the introduction of a uncorecache hierarchy. This means that when a guaranteed container requires CPUs that are equal to or greater than the size of a NUMA or socket, the CPU allocation will behave as usual and schedule the full socket or NUMA. The scheduled CPUs will be subtracted from the quantity of CPUs required for the container.
+
+When/once the required CPUs for a container are smaller than the size of a NUMA, the algorithm will be implemented as follows:
+1. Scan each socket in numerical order. If required CPUs is less than the size of available CPUs on the socket, pick this socket.
+2. Within the chosen socket, scan each NUMA in numerical order. If the required CPUs is less than the size of the available CPUs on the NUMA, pick this NUMA.
+3. Within the chosen NUMA, scan through every uncore cache index in numerical order:
+   - If the required CPUs is greater than or equal the size of the **total** amount of CPUs to an uncore cache, assign the full uncore cache CPUs if they are all **unscheduled** within the group. Subtract the amount of scheduled CPUs from the quantity of required CPUs. 
+     - Continue scanning uncore caches in numerical order and assigning full uncore cache CPU groups until the required quantity is less than the **total** number of CPUs to a uncore cache group.
+   - If the required CPUs is less than the size of the **total** amount of CPUs to an uncore cache, scan through each uncore cache index in numerical order starting from the first index, and assign CPUs if there are enough **available** CPUs within the uncore cache group.
+4. If the required amount of CPUs cannot fit within an uncore cache group and there are enough schedulable CPUs on the node, assign cores in numerical order.
```

Comment:
so, this option will be "prefrence" on alignment, not the "requirement". Probably good to emphasise in the initial description.

Reply:
Add this to the summary and the design details

Reply:
Yes, I agree I will be adding this